tests/core/checks/custom/BUILD.bazel

```diff
@@ -0,0 +1,150 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+load("@io_bazel_rules_go//tests:bazel_tests.bzl", "bazel_test")
+
+_workspace = """
+load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains")
+
+go_rules_dependencies()
+go_register_toolchains(go_checker="@//:my_checker")
+"""
+
+_single_check_build = """
+load("@io_bazel_rules_go//go:def.bzl", "go_checker", "go_tool_library")
+
+go_checker(
+    name = "my_checker",
+    deps = [":foo_func_check"],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "foo_func_check",
+    srcs = ["foo_func_check.go"],
+    importpath = "foofunccheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+"""
+
+_multiple_checks_build = """
+load("@io_bazel_rules_go//go:def.bzl", "go_checker", "go_tool_library")
+
+go_checker(
+    name = "my_checker",
+    deps = [
+        ":foo_func_check",
+        ":return_bool_check",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "foo_func_check",
+    srcs = ["foo_func_check.go"],
+    importpath = "foofunccheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+
+go_tool_library(
+    name = "return_bool_check",
+    srcs = ["return_bool_check.go"],
+    importpath = "returnboolcheck",
+    deps = ["@io_bazel_rules_go//go/tools/analysis:analysis"],
+    visibility = ["//visibility:public"],
+)
+"""
+
+_no_error_check = """
+if [[ result -eq 1 ]]; then
+  echo "error: unexpected build error" >&2
+  result=1
+else
+  result=0
+fi
+"""
+
+_has_err_check_tmpl = """
+if [[ result -eq 0 ]]; then
+  echo "error: expected build error" >&2
+  result=1
+else
+  {check_err}
+fi
+"""
+
+_check_err_tmpl = """
+  grep -q '{err}' bazel-output.txt
```

Comment:
Nit: simplify to `if ! grep -q '{err}' bazel-output.txt; then`

Reply:
Done.